src/main/java/com/google/api/codegen/configgen/transformer/MethodTransformer.java

```diff
@@ -48,6 +47,15 @@ public MethodTransformer(PagingParameters pagingParameters) {
     this.pagingParameters = pagingParameters;
   }
 
+  /** Get the ResourceNameTreatment for a method. */
+  @Nullable
+  abstract ResourceNameTreatment getResourceNameTreatment(MethodModel methodModel);
+
+  /** Make the page streaming response view for a method. */
+  @Nullable
+  abstract PageStreamingResponseView generatePageStreamingResponse(
+      PagingParameters pagingParameters, MethodModel method);
```

Comment:
I strongly prefer not to use multi-level inheritance with implementations at multiple levels; it makes code flow hard to follow.

Reply:
Moved the pagingParameters to the subclasses, so this abstract base class doesn't have it as a constructor parameter.